WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dashboard/CardsSourceTest.kt

```diff
@@ -299,12 +324,13 @@ class CardsSourceTest : BaseUnitTest() {
         cardSource.build(testScope(), SITE_LOCAL_ID).observeForever { }
 
         cardSource.refresh()
+        advanceUntilIdle()
 
         assertThat(result.size).isEqualTo(5)
         assertThat(result[0]).isFalse // init
         assertThat(result[1]).isTrue // build(...) -> refresh()
-        assertThat(result[2]).isFalse // build(...) -> cardsStore.fetchCards(...) -> error
-        assertThat(result[3]).isTrue // refresh()
+        assertThat(result[2]).isTrue // build(...) -> cardsStore.fetchCards(...) -> error
```

Comment:
👋 @RenanLukas , thank you for your comment and for being detailed on this! 🥇 

This `Coroutines` update introduced a major shift in how our tests are structured. I tried to make those as straightforward as possible and also consistent with WCAndroid.

-----

The most interesting part that you will notice is that I updated the `MainCoroutineScopeRule` (now named as `CoroutineTestRule`). And, as part of [this](https://github.com/wordpress-mobile/WordPress-Android/pull/17673/commits/e6f3f89e9ad8d7496be8e4779c8ad70fe7ada40d) update, you will notice that the `finished(...)` setup got updated from the below:

```
    override fun finished(description: Description?) {
        super.finished(description)
        advanceUntilIdle()
        cleanupTestCoroutines()
        Dispatchers.resetMain()
    }
```

And into the below:

```
    override fun finished(description: Description) {
        super.finished(description)
        Dispatchers.resetMain()
    }
```

As such, the below two extra configuration lines that were previously applied to all test got removed:

```
        advanceUntilIdle()
        cleanupTestCoroutines()
```

Thus, you also saw so many fix commits and fixed tests, where I had to manually add the `advanceUntilIdle()`.

-----

Having explained the above, my reasoning is that this caused the need for small tweaks like that.

Also, since I was the author of this `CardsSourceTest.kt` and knowing that I introduced this long list of asserts, per result, I know that they are not that easy to reason about as they were actually added automatically. I only added those extra asserts to make sure that when the main code gets updated, if the logic get unintentionally updated, those should warn the author of that change so that they are aware of it and so that they make sure that the update they did on the logic was intentional. Also, you will notice the comments per assert, those were added to make sure we can reason about the flow, but that's the extend of it, and actually, might even be outdated by now... 🤷 

As such, and since this PR only updates the testing logic, I didn't worry too much about this change, that's why I ended up feeling comfortable to just update those values and make sure these tests still pass without adding any extra effort to verify whether the `1.6.4` update affected the main code. That's why I did and asked for a thorough test on this PR, so that we can verify its correctness main-code-wise as nothing else was done coding-wise there.

Does this makes sense? 🤔

PS: The same applies to `BloggingPromptCardSourceTest.kt` as I am seeing those tests got copy-pasted.

Reply:
Yes, makes sense. Thanks for the detailed explanation @ParaskP7 .